test/minikube/minikube_domain_resync_test.go

```diff
@@ -188,6 +194,23 @@ func checkInitialMembership(t assert.TestingT, configJson string, expectedSize i
 	assert.Equal(t, expectedSize, len(adminConfig.InitialMembership))
 }
 
+func modifyKubeInspectorRoleInPlace(t *testing.T, modificationFunc func(role *rbacv1.Role)) {
+	inspectorRoleFile := filepath.Join(testlib.ADMIN_HELM_CHART_PATH, "templates", "role.yaml")
+	originalData, err := ioutil.ReadFile(inspectorRoleFile)
+	assert.NoError(t, err)
+	testlib.AddTeardown(testlib.TEARDOWN_ADMIN, func() { ioutil.WriteFile(inspectorRoleFile, originalData, 0644) })
+
+	output := helm.RenderTemplate(t, &helm.Options{}, testlib.ADMIN_HELM_CHART_PATH, "release-name", []string{"templates/role.yaml"})
+	roles := testlib.SplitAndRenderRole(t, output, 1)
+	modificationFunc(&roles[0])
+	roleBytes, err := yaml.Marshal(&roles[0])
+	assert.NoError(t, err)
+	err = ioutil.WriteFile(inspectorRoleFile, roleBytes, 0644)
+	assert.NoError(t, err)
+	out, _ := ioutil.ReadFile(inspectorRoleFile)
```

Comment:
are you overriding the file on disk? I would just create a temp file

Reply:
As I know helm will send all files in the `templates` directory via the rendering engine. Either we should edit the file in place and revert the change or we should move the original file and revert it later. I preferred the first option as we don't need to write a lot of helper code just to move files around.